translib/transformer/sw_portchannel.go

```diff
@@ -0,0 +1,451 @@
+	if log.V(3) {
+		log.Info("Entering DbToYang_lag_min_links_xfmr")
+	}
+	var err error
+	result := make(map[string]interface{})
+
+	err = validatePortChannel(inParams.d, inParams.key)
+	if err != nil {
+		log.Infof("DbToYang_lag_min_links_xfmr Error: %v ", err)
+		return result, err
+	}
+	data := (*inParams.dbDataMap)[inParams.curDb]
+	links, ok := data[PORTCHANNEL_TABLE][inParams.key].Field["min_links"]
+	if ok {
+		linksUint16, err := uint16Conv(links)
+		if err != nil {
+			return result, err
+		}
+		result["min-links"] = linksUint16
+	} else {
+		if log.V(3) {
+			log.Info("min-links set to 1 (default value)")
+		}
+		linksUint16, err := uint16Conv(DEFAULT_PORTCHANNEL_MIN_LINKS)
+		if err != nil {
+			return result, err
+		}
+		result["min-links"] = linksUint16
+	}
+
+	return result, err
+}
+
+// DbToYang_intf_lag_state_xfmr is a DB to Yang translation overloaded method for PortChannel GET operation
+var DbToYang_intf_lag_state_xfmr SubTreeXfmrDbToYang = func(inParams XfmrParams) error {
+	var err error
+
+	intfsObj := getIntfsRoot(inParams.ygRoot)
+	if intfsObj == nil || intfsObj.Interface == nil {
+		errStr := "Failed to Get root object!"
+		log.Errorf(errStr)
+		return errors.New(errStr)
+	}
+	pathInfo := NewPathInfo(inParams.uri)
+	ifName := pathInfo.Var("name")
+	if _, ok := intfsObj.Interface[ifName]; !ok {
+		obj, _ := intfsObj.NewInterface(ifName)
+		ygot.BuildEmptyTree(obj)
+	}
+	intfObj := intfsObj.Interface[ifName]
+	if intfObj.Aggregation == nil {
+		ygot.BuildEmptyTree(intfObj)
+	}
+	if intfObj.Aggregation.State == nil {
+		ygot.BuildEmptyTree(intfObj.Aggregation)
+	}
+	intfType, _, err := getIntfTypeByName(ifName)
+	if intfType != IntfTypePortChannel || err != nil {
+		intfTypeStr := strconv.Itoa(int(intfType))
+		errStr := "TableXfmrFunc - Invalid interface type: " + intfTypeStr
+		log.Warning(errStr)
+		return errors.New(errStr)
+	}
+	/*Validate given PortChannel exists */
+	err = validatePortChannel(inParams.d, ifName)
+	if err != nil {
+		return err
+	}
+
+	targetUriPath := pathInfo.YangPath
+	log.Info("targetUriPath is ", targetUriPath)
+	lagInfoMap := make(map[string]db.Value)
+	ocAggregationStateVal := intfObj.Aggregation.State
+	err = fillLagInfoForIntf(inParams, inParams.d, &ifName, lagInfoMap, &targetUriPath)
+	if err != nil {
+		log.Errorf("Failed to get info: %s failed!", ifName)
+		return err
+	}
+	log.Info("Succesfully completed DB map population!", lagInfoMap)
+	switch targetUriPath {
+	case "/openconfig-interfaces:interfaces/interface/openconfig-if-aggregate:aggregation/state/min-links":
+		log.Info("Get is for min-links")
+		attr := "min-links"
+		err = getLagStateAttr(&attr, &ifName, lagInfoMap, ocAggregationStateVal)
+		if err != nil {
+			return err
+		}
+	case "/openconfig-interfaces:interfaces/interface/aggregation/state":
+		fallthrough
+	case "/openconfig-interfaces:interfaces/interface/openconfig-if-aggregate:aggregation/state":
```

Comment:
URI at line 343 and 345 represent the same path. Isn't the URI normalized in the infra so that multiple cases are not needed? If yes, then we need only one cases here.

Reply:
Checked and the one without the module-prefix was unnecessary. Removed. Thank you for the suggestion!